src/ar_firewall.erl

```diff
@@ -1,71 +1,144 @@
 -module(ar_firewall).
--export([start/0, scan_tx/2]).
+-export([start/0, reload/0, scan_tx/1]).
 -include("ar.hrl").
 -include("av/av_recs.hrl").
 -include_lib("eunit/include/eunit.hrl").
 
 %%% Arweave firewall implementation.
 
-%% State definition.
-%% Should store compiled binary scan objects.
+%% Stores compiled binary scan objects.
 -record(state, {
-	sigs = [] % the set of known signatures to filter against
+	tx_id_sigs = [], % a set of known signatures to filter transaction identifiers against
+	content_sigs = [] % a set of known signatures to filter transaction content against
 }).
 
 %% @doc Start a firewall node.
 start() ->
-	spawn(
-		fun() ->
-			server(#state {
-				sigs = av_sigs:all()
-			})
-		end
-	).
+	Firewall = whereis(default_firewall),
+	case Firewall of
+		undefined ->
+			F = spawn(
+				fun() ->
+					server(#state {
+						tx_id_sigs = av_sigs:load(ar_meta_db:get(transaction_blacklist)),
+						content_sigs = av_sigs:load(ar_meta_db:get(content_policies))
+					})
+				end
+			),
+			erlang:register(default_firewall, F);
```

Comment:
I see no reason to use a different name than `ar_firewall`. It's a singleton which you cannot start more of, so it makes sense to use `?MODULE`.

Reply:
✅